File: api/src/main/java/io/bunnies/baas/services/RequestTrackerSingleton.java
package io.bunnies.baas.services;

import io.bunnies.baas.resources.BunnyResources;
import redis.clients.jedis.JedisPool;

public class RequestTrackerSingleton {
    private static RequestTracker requestTracker;

    public static RequestTracker getInstance(JedisPool jedisPool, BunnyResources resources) {
        if (requestTracker == null) {
            requestTracker = new RequestTracker(jedisPool, resources);
        }

        return requestTracker;
    }
}

File: api/src/main/java/io/bunnies/baas/services/RequestTrackerSingleton.kt
package io.bunnies.baas.services

import io.bunnies.baas.resources.BunnyResources
import redis.clients.jedis.JedisPool

object RequestTrackerSingleton {
    private var requestTracker: RequestTracker? = null

    fun getInstance(jedisPool: JedisPool, resources: BunnyResources): RequestTracker {
        return requestTracker ?: createNewRequestTracker(jedisPool, resources)
    }

    private fun createNewRequestTracker(jedisPool: JedisPool, resources: BunnyResources): RequestTracker {
        val requestTracker = RequestTracker(jedisPool, resources)
        this.requestTracker = requestTracker

        return requestTracker
    }
}